src/Common/TaskIO.php

```diff
@@ -99,7 +101,18 @@ protected function printTaskError($text, $context = null)
      */
     protected function printTaskDebug($text, $context = null)
     {
-        $this->logger()->debug($text, $this->getTaskContext($context));
+        $this->printTaskOutput(LogLevel::DEBUG, $text, $this->getTaskContext($context));
+    }
+
+    private function printTaskOutput($level, $text, $context)
+    {
+        if ($this instanceof ProgressIndicatorAwareInterface) {
+            $this->hideProgressIndicator();
+        }
+        $this->logger()->log($level, $text, $this->getTaskContext($context));
+        if ($this instanceof ProgressIndicatorAwareInterface) {
+            $this->showProgressIndicator();
+        }
```

Comment:
this thing was confusing to me and probably it is in general. I'd replace it with something like this:

```php
$inProgress = false;
if ($this instanceof ProgressIndicatorAwareInterface) {
  $inProgress = $this->inProgress();
}

if ($inProgress) {    
 $this->hideProgressIndicator();
}

$this->logger()->log($level, $text, $this->getTaskContext($context));

if ($inProgress) {
 $this->showProgressIndicator();
}
```